lib/ace/mode/javascript_highlight_rules.js

```diff
@@ -69,7 +69,7 @@ var JavaScriptHighlightRules = function(options) {
     }, "identifier");
 
     // keywords which can be followed by regular expressions
-    var kwBeforeRe = "case|do|else|finally|in|instanceof|return|throw|try|typeof|yield|void";
+    var kwBeforeRe = "case|do|else|finally|in|of|instanceof|return|throw|try|typeof|yield|void";
```

Comment:
It's better to not add `of` here, since it is a contextual keyword and something like `for (i of /foo/)` is less likely to occur than something like `of/7`